pkg/resources/envoy/configmap.go

```diff
@@ -64,11 +64,7 @@ func GenerateEnvoyConfig(kc *v1beta1.KafkaCluster, envoyConfig *v1beta1.EnvoyCon
 	var clusters []*envoyapi.Cluster
 
 	for _, brokerId := range util.GetBrokerIdsFromStatus(kc.Status.BrokersState, log) {
-		broker := util.GetBrokerSpecFromId(kc.Spec, int32(brokerId), log)
```

Comment:
this seems to be a bug in upstream master right? I see they are still trying to look up broker in spec. can you comment?

Reply:
This is part of the previous rebase. and that's the reason of this PR 😄 
The previous implementation (for the envoy multi-zone) had this link between the Kafka cluster Spec and the Envoy deployment. Since the Spec is updated sooner than the brokers become available (or are removed), this was resulting in a bug during Kafka scale up / down. Now, we `GetBrokerConfigGroupFromStatus` which is a lot more accurate.